velox/exec/HashBuild.cpp

```diff
@@ -689,14 +692,18 @@ bool HashBuild::finishHashBuild() {
   otherTables.reserve(peers.size());
   SpillPartitionSet spillPartitions;
   Spiller::Stats spillStats;
-  if (!antiJoinHasNullKeys_) {
+  if (joinHasNullKeys_ && isNullAwareAntiJoin(joinType_)) {
+    joinBridge_->setAntiJoinHasNullKeys();
+  } else {
     for (auto& peer : peers) {
       auto op = peer->findOperator(planNodeId());
       HashBuild* build = dynamic_cast<HashBuild*>(op);
       VELOX_CHECK(build);
-      if (build->antiJoinHasNullKeys_) {
-        antiJoinHasNullKeys_ = true;
-        break;
+      if (build->joinHasNullKeys_) {
+        joinHasNullKeys_ = true;
+        if (isNullAwareAntiJoin(joinType_)) {
```

Comment:
Shall we call joinBridge_->setAntiJoinHasNullKeys() here? Is it a bug in current code? Thanks!

Reply:
This is indeed a pre-existing bug. I'm going add `joinBridge_->setAntiJoinHasNullKeys();`.